src/libraries/System.Drawing.Common/ref/System.Drawing.Common.csproj

```diff
@@ -12,6 +12,7 @@
     <ProjectReference Include="..\..\System.Collections.NonGeneric\ref\System.Collections.NonGeneric.csproj" />
     <ProjectReference Include="..\..\System.ComponentModel\ref\System.ComponentModel.csproj" />
     <ProjectReference Include="..\..\System.ComponentModel.Primitives\ref\System.ComponentModel.Primitives.csproj" />
+    <ProjectReference Include="..\..\System.ComponentModel.TypeConverter\ref\System.ComponentModel.TypeConverter.csproj" />
```

Comment:
Hmm, @ericstj we've been avoiding bringing `System.ComponentModel.TypeConverter` closure into `System.Drawing.Common`, what do you suggest here, maybe pushing these two attributes down to primitives or `S.ComponentModel`?

Reply:
I believe the attribute usage is decoupled enough from the normal use case that we should be able to add the dependency and avoid loading TypeConverter, as I mentioned here: https://github.com/dotnet/runtime/issues/1011#issuecomment-592660163

Reply:
@eerhardt what level of trimming would apps need so that TypeConverter is trimmed from the closure if we only use attributes?

Reply:
They would need to pass `--used-attrs-only true` (which is not the default) to the linker in order to get attributes that aren't being read to be trimmed.

See https://github.com/mono/linker/blob/7573e99c7665f565fe4a6ac4784d5c641213d5e1/src/linker/Linker/Driver.cs#L960

Reply:
Note: The `--used-attrs-only` setting has issues with it, as called out in https://github.com/mono/linker/issues/952#issuecomment-591984078. So I wouldn't expect high usage of this setting until those problems are fixed.

Reply:
@ericstj are we ok with this given that information then?

Reply:
I'll defer to @eerhardt to understand if this attribute usage matters for the customer facing trimming scenario.  I believe our goal should be that in type-trimming case that a customer app that doesn't use TypeConverter should not bring it into their app.  I would hope that remains true for both this change and for the fix to #1011

Reply:
@safern is there any further updates on the review of this PR?

Reply:
> I would hope that remains true for both this change and for the fix to #1011

@eerhardt do you know if that's true?

Reply:
Today, the customer facing trimming only trims unused assemblies as a whole, and not individual types/methods/etc. (From my understanding)

In .NET 5 we plan to support trimming unused types/methods with the priority being mobile apps like Xamarin and Blazor. I'm not sure if we are going to target Desktop/Console/Web applications with this type of trimming. @vitek-karas - do you know if this is a committed scenario in .NET 5?

Reply:
AFAIK we will not block it in any scenario, but it's not a goal (right now) outside of Xamarin/Blazor. So it's likely to not work very well (more likely to work on simpler console apps, much less likely to work well on Web or GUI apps).

Reply:
@safern is the "trimming" the same context as in this [article](https://github.com/mono/linker/blob/master/src/ILLink.Tasks/README.md)? Do I need to update something else to add to this PR?

Reply:
@eriawan yeah. What @ericstj and I discussed was testing the trimming behavior with this change and based on the outcome decide if we want to take the change as is. Thanks for offering action, if you want to try that, feel free and share your findings as well 😄 . I'll find time later today to test this.

Reply:
@eriawan I haven't gotten a chance to do this test. Would it be possible for you to do it? 

You have to build the System.Drawing.Common package -- here's an example on how to build a simple package: https://github.com/dotnet/runtime/issues/32249#issuecomment-622124361

Then write a console app that references 

`PrintDocument` and then run `dotnet publish -r win-x64` with `/p:PublishTrimmed=true` and see if the resulting directory contains `System.ComponentModel.TypeConverter` in there and if these attributes are preserved on `PrintDocument`?

Reply:
@eerhardt @vitek-karas is that the right way to validate this?

Reply:
@safern yes, that should do it

Reply:
@safern 
Thanks for the clue and the info to test! 👍 
The [current ILLink.Tasks documentation on the link I have shared](https://github.com/mono/linker/blob/master/src/ILLink.Tasks/README.md) is not quite clear in the sense of how exactly the expected result should be, because I have tried to read and try to use `PublishTrimmed` based on the documentation, 

Your suggested instruction is easier to understand, I'm going to try this now.

Reply:
@eriawan did you get a change to try this?

Reply:
@safern 
ok, I've compiled and I got the resulting System.Drawing.Common.5.0.0-dev.nupkg. 
Then I configure a local nuget feed on my machine, add the local feed's folder path as new nuget source.

I then created a new console project using .NET Core 5.0 Preview 5
After I  executed `dotnet publish -r win-x64 /p:PublishTrimmed=true` on the project folder, I get these warnings:

![image](https://user-images.githubusercontent.com/8773147/84943708-f8cc0480-b10e-11ea-940e-b34314816c56.png)

The text below is the copied text from the screenshot:
```
D:\DOTNET_CORE_CONTRIBUTION_SANDBOX\netcore5.0>dotnet publish -r win-x64 /p:PublishTrimmed=true SystemDrawingTrimTest\SystemDrawingTrimTest.csproj
Microsoft (R) Build Engine version 16.7.0-preview-20258-02+26f6d1d87 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
D:\DOTNET_CORE_CONTRIBUTION_SANDBOX\netcore5.0\SystemDrawingTrimTest\SystemDrawingTrimTest.csproj : warning NU1603: System.Drawing.Common 5.0.0-dev depends on Microsoft.NETCore.Platforms (>= 5.0.0-dev) but Microsoft.NETCore.Platforms 5.0.0-dev was not found. An approximate best match of Microsoft.NETCore.Platforms 5.0.0-preview.1.20120.5 was resolved.
D:\DOTNET_CORE_CONTRIBUTION_SANDBOX\netcore5.0\SystemDrawingTrimTest\SystemDrawingTrimTest.csproj : warning NU1603: System.Drawing.Common 5.0.0-dev depends on Microsoft.Win32.SystemEvents (>= 5.0.0-dev) but Microsoft.Win32.SystemEvents 5.0.0-dev was not found. An approximate best match of Microsoft.Win32.SystemEvents 5.0.0-preview.1.20120.5 was resolved.
  Restored D:\DOTNET_CORE_CONTRIBUTION_SANDBOX\netcore5.0\SystemDrawingTrimTest\SystemDrawingTrimTest.csproj (in 1.85 min).
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
D:\DOTNET_CORE_CONTRIBUTION_SANDBOX\netcore5.0\SystemDrawingTrimTest\SystemDrawingTrimTest.csproj : warning NU1603: System.Drawing.Common 5.0.0-dev depends on Microsoft.NETCore.Platforms (>= 5.0.0-dev) but Microsoft.NETCore.Platforms 5.0.0-dev was not found. An approximate best match of Microsoft.NETCore.Platforms 5.0.0-preview.1.20120.5 was resolved.
D:\DOTNET_CORE_CONTRIBUTION_SANDBOX\netcore5.0\SystemDrawingTrimTest\SystemDrawingTrimTest.csproj : warning NU1603: System.Drawing.Common 5.0.0-dev depends on Microsoft.Win32.SystemEvents (>= 5.0.0-dev) but Microsoft.Win32.SystemEvents 5.0.0-dev was not found. An approximate best match of Microsoft.Win32.SystemEvents 5.0.0-preview.1.20120.5 was resolved.
  SystemDrawingTrimTest -> D:\DOTNET_CORE_CONTRIBUTION_SANDBOX\netcore5.0\SystemDrawingTrimTest\bin\Debug\net5.0\win-x64\SystemDrawingTrimTest.dll
  Optimizing assemblies for size, which may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  SystemDrawingTrimTest -> D:\DOTNET_CORE_CONTRIBUTION_SANDBOX\netcore5.0\SystemDrawingTrimTest\bin\Debug\net5.0\win-x64\publish\
```

Then I look at the content of publish folder. The System.ComponentModer.Converter.dll is available there.
This is the folder content:

![image](https://user-images.githubusercontent.com/8773147/84944501-0d5ccc80-b110-11ea-867e-c8a0e4e3a83d.png)

This is the content of SystemDrawingTrimTest.csproj:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="System.Drawing.Common" Version="5.0.0-dev" />
  </ItemGroup>

</Project>

```

This is the content of Program.cs:
```
using System;
using System.Drawing.Printing;

namespace SystemDrawingTrimTest
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello World!");
            var pd = new PrintDocument();
            Console.WriteLine(pd.GetType().ToString());
        }
    }
}

```

So, what should we do next? Also please let me know if I miss something .

Reply:
@safern 
Just to remind, please review my previous report about trying trim on System.Drawing.Common. 
thanks before! 🙂

Reply:
Thanks, @eriawan. 

@eerhardt @ericstj so it seems with the current change this will cause people to have System.ComponentModel.TypeConverter in their standalone apps even when publishing trimmed. Any recommendations to avoid that?

Reply:
@safern @eerhardt @ericstj 
pardon, is there any update about the recommendation to avoid incllusion of System.ComponentModel.TypeConverter assembly using trimmed publish?

Reply:
Well, we could push the attribute types down to a lower assembly.  Either System.ComponentModel.Primitives or System.ObjectModel as @safern suggested.  It's still not clear to me how important this is.  One thing we could do is take this as is and then if we get feedback on app size issues (eg: if this matters for browser) then we push the types down.  The thing that I'm trying to reason about is how much we care about this metric, certainly in Winforms use case we don't since both TypeConverter and Drawing will always be used.

Reply:
@ericstj fair point; we can do that. Will reset CI then to make sure we get a fresh run and then merge.